nemo/lightning/io/pl.py

```diff
@@ -174,6 +195,11 @@ def load_checkpoint(
         if not fs.isdir(path):
             raise ValueError(f"Distributed checkpoints should be a directory. Found: {path}.")
 
+        # Load from ckpt_path/weights (new format) if it exists
+        path = ckpt_to_weights_subdir(path)
+        if hasattr(path, "base_model_path") and not path.base_model_path.exists():
```

Comment:
is the assumption that `path.base_model_path.parts[-1] == "weights"` here? if yes, can we include this as an assert inside the if's block?

Reply:
@cuichenx can you confirm whether this should always be the case?

Reply:
> is the assumption that path.base_model_path.parts[-1] == "weights" here?
Yes, because `ckpt_to_weights_subdir` adds /weights to the path

> can we include this as an assert inside the if's block?
I don't understand why we want to do this? This block is to account for the old checkpoint format where imported checkpoints don't have the weight/context folder

Reply:
The goal of my comment to include `assert` statements was to make sure that there's some code that checks for the assumed path structure (whatever that might be), mostly for testing purposes.